src/dstore/pmix_esh.h

```diff
@@ -26,6 +26,23 @@ BEGIN_C_DECLS
 
 #define PMIX_DSTORE_ESH_BASE_PATH "PMIX_DSTORE_ESH_BASE_PATH"
 
+#define PTHREAD_LOCK_ENABLE 1
+#define FCNTL_LOCK_ENABLE   1
```

Comment:
Could we make these `configure` options? Is there any reason someone would want to choose the `flock` version over the `pthread_lock`? Or should we just remove the `flock` code?